### PR TITLE
OPENEUROPA-1325: Media entity browser handling for files

### PR DESCRIPTION
**Pull request by sergepavle**
### Description

Media entity browser handling for files

### Change log

- [x]  Create entity browser widget for "creating" file media entities
- [x]  Adjust the test that selects media via the entity browser to select also existing file entities
- [x]  Test the entity browser widget that creates and selects the file entity

### CODE REVIEW

File: modules/oe_media_demo/config/install/entity_browser.browser.media_entity_browser.yml [hunk 1]
```diff
@@ -49,3 +49,13 @@ widgets:
     weight: 3
     label: 'Add Image'
     id: entity_form
+  e9f8e84a-9574-4de7-90de-caec7acf030d:
```
Comment (imanoleguskiza) [inline modules/oe_media_demo/config/install/entity_browser.browser.media_entity_browser.yml]: Why do we move this browser definition from the main module to the demo one? This doesn't seem to be the scope of the ticket

File: tests/features/media-file.feature [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    And I fill in "Title" with "Media demo"
+    And I click fieldset "Media browser field" of entity browser widget
+    When I press the "Select entities" button
+    Then I should see entity browser modal window
+    When I click "View"
+    And I select the "Media document" media entity in the entity browser modal window
+    And I press the "Select entities" button
+    And I press the "Save" button
+    Then I should see "sample.pdf"
+
```
Comment (imanoleguskiza) [inline tests/features/media-file.feature]: Please remove extra empty spaces

File: modules/oe_media_demo/config/install/entity_browser.browser.media_entity_browser.yml (hunk 1 above)
Comment (sergepavle) [reply modules/oe_media_demo/config/install/entity_browser.browser.media_entity_browser.yml]: This is decision was discussed with @upchuk . End users (site builders) could have many field which allow use only some media types. That is why common media entity browser which support all existing media type could be not relevant anymore.

File: tests/features/media-file.feature (hunk 2 above)
Comment (sergepavle) [reply tests/features/media-file.feature]: fixed

File: modules/oe_media_demo/config/install/entity_browser.browser.media_entity_browser.yml (hunk 1 above)
Comment (imanoleguskiza) [reply modules/oe_media_demo/config/install/entity_browser.browser.media_entity_browser.yml]: Fair enough! :)

File: tests/features/media-file.feature [hunk 3]
```diff
@@ -2,31 +2,33 @@
...
+    And I click the fieldset "Media browser field"
     And I press the "Select entities" button
     Then I should see entity browser modal window
     When I click "View"
     And I select the "Media document" media entity in the entity browser modal window
     And I press the "Select entities" button
     And I press the "Save" button
-    Then I should see "sample.pdf"
+    Then I should see the link "sample.pdf"
+    # Cleanup of the media entity.
```
Comment (voidtek) [inline tests/features/media-file.feature]: The cleanup should be done with @afterScenario.
It's not a blocker but please create a follow-up ticket if you don't take this change it here.

**Pull request merged**